omod/src/main/webapp/resources/scripts/visit/visit.js

```diff
@@ -631,6 +631,43 @@ angular.module("visit", [ "filters", "constants", "encounterTypeConfig", "visitS
                 });
             }
 
+            function loadAllPatientEncounters(patientUuid) {
+                let encounters = [];
+                const encounterRequestArgs = {
+                    patient: patientUuid,
+                    encounterType: encounterTypeUuid ? encounterTypeUuid : '',
+                    order: 'desc',
+                    v: "custom:(encounterDatetime,encounterType:(uuid,display))"
+                };
+                Encounter.get(encounterRequestArgs).$promise.then(function(response) {
+                    encounters = transformEncounterData(response);
+                    getMoreEncountersIfPresent(response).then(function() {
+                        $scope.visit.allEncounters = encounters
+                    });
+                });
+                function getMoreEncountersIfPresent(response) {
+                    const nextLink = response.links && response.links.find(l => l.rel === "next");
+                    if (nextLink) {
+                        return $http.get(nextLink.uri).then(function(newResponse) {
+                            const data = newResponse.data;
+                            encounters = encounters.concat(transformEncounterData(data));
+                            return getMoreEncountersIfPresent(data);
```

Comment:
I checked that this recursion works by setting the default limit to 100 and going to a patient with 803 encounters.

Reply:
Nice, looks good.